src/pd/cluster.rs

```diff
@@ -191,9 +195,10 @@ impl Connection {
     }
 
     // Re-establish connection with PD leader in synchronized fashion.
-    pub fn reconnect(
+    pub async fn reconnect(
         &self,
-        old_cluster: &Cluster,
+        cluster_id: u64,
+        previous_members: &pdpb::GetMembersResponse,
```

Comment:
> Why make this change? It seems to expose more of `Cluster`'s implementation.

In previous version, `old_cluster` is actually  a `RwLockReadGuard<Cluster>`,which can not pass through async barrier, and now this method is a async method.

Reply:
> I don't understand why we can pass `previous_members`, but not `old_cluster`, istm they are both references with the same lifetimes?

Maybe a move here is better than reference, I changed the parameter to move, `previous_members` is cloned from a field of `old_cluster`, which is just a list of data, `old_cluster` is a more complex struct, refactory it to make it possible to clone will be tricky.I must clone because because `old_cluster` is protected by a `RwLockReadGuard` which cannot pass through a sync barrier(which is `async fn` here )